pkg/minikube/cruntime/docker.go

```diff
@@ -580,12 +580,17 @@ func (r *Docker) configureDocker(driver string) error {
 		},
 		StorageDriver: "overlay2",
 	}
-	if r.GPUs {
-		assets.Addons["nvidia-device-plugin"].EnableByDefault()
-		daemonConfig.DefaultRuntime = "nvidia"
-		runtimes := &dockerDaemonRuntimes{}
-		runtimes.Nvidia.Path = "/usr/bin/nvidia-container-runtime"
-		daemonConfig.Runtimes = runtimes
+	if r.GPUs != "" {
+		if r.GPUs == "all" || r.GPUs == "nvidia" {
+			assets.Addons["nvidia-device-plugin"].EnableByDefault()
+			daemonConfig.DefaultRuntime = "nvidia"
+			runtimes := &dockerDaemonRuntimes{}
+			runtimes.Nvidia.Path = "/usr/bin/nvidia-container-runtime"
```

Comment:
I'm not sure if `--gpus=all` will work for AMD, have you tried it? Because it's going to set the runtime to use the Nvidia runtime, and if there's no Nvidia card I assume it's going to fail.

Reply:
I was thinking here if for some reason you had both nvidia and amd, its unlikely, I think best to just activate amd if gpus is set to amd.